python/vpc_flow_logs_enabled.py

```diff
@@ -0,0 +1,50 @@
+#
+# This file made available under CC0 1.0 Universal (https://creativecommons.org/publicdomain/zero/1.0/legalcode)
+#
+
+import boto3
+import json
+
+def evaluate_compliance(config_item, is_flow_logs_enabled):
+    if (config_item['resourceType'] != 'AWS::EC2::VPC'):
+        return 'NOT_APPLICABLE'
+
+    elif is_flow_logs_enabled:
+        return 'COMPLIANT'
+    else:
+        return 'NON_COMPLIANT'
+
+def is_flow_logs_enabled(vpc_id):
+    ec2 = boto3.client('ec2')
+    response = ec2.describe_flow_logs(
+        Filter=[
+            {
+                'Name': 'resource-id',
+                'Values': [
+                    vpc_id,
+                ]
+            },
+        ],
+    )
+    if len(response[u'FlowLogs']) != 0: return True
+
+def lambda_handler(event, context):
+    invoking_event = json.loads(event['invokingEvent'])
+
+    compliance_value = 'NOT_APPLICABLE'
+
+    vpc_id = invoking_event['configurationItem']['resourceId']
+    compliance_value = evaluate_compliance(
+        invoking_event['configurationItem'], is_flow_logs_enabled(vpc_id))
```

Comment:
The way this is written, is_flow_logs_enabled will be called for all configurationItems passed to this lambda_handler, since it is evaluated before the outer function (evaluate_compliance) is called. Since  is_flow_logs_enabled does a ec2.describe_flow_logs service call each time, this could be quite inefficient. Suggest refactoring code to explicitly test for applicability first before calling is_flow_logs_enabled(vpc_id)